src/gui/editors/AutomationEditor.cpp

```diff
@@ -2238,7 +1844,62 @@ void AutomationEditor::updateTopBottomLevels()
 
 
 
+/* @brief Given a mouse coordinate, returns a timeMap::iterator that points to
+ *        the first node inside a square of side "r" pixels from those
+ *        coordinates. In simpler terms, returns the automation node on those
+ *        coordinates.
+ * @param Int X coordinate
+ * @param Int Y coordinate
+ * @param Boolean. True to check if the outValue of the node was clicked instead
+ *        (defaults to false)
+ * @param Int R distance in pixels
+ * @return timeMap::iterator with the clicked node, or timeMap.end() if none was clicked.
+ */
+AutomationEditor::timeMap::iterator AutomationEditor::getNodeAt(int x, int y, bool outValue /* = false */, int r /* = 5 */)
+{
+	QMutexLocker m(&m_patternEditorMutex);
 
+	// Remove the VALUES_WIDTH from the x position, so we have the actual viewport x
+	x -= VALUES_WIDTH;
+	// Convert the x position to the position in ticks
+	int posTicks = (x * MidiTime::ticksPerBar()/m_ppb) + m_currentPosition;
+
+	// Get our pattern timeMap and create a iterator so we can check the nodes
+	timeMap & tm = m_pattern->getTimeMap();
+	timeMap::iterator it = tm.begin();
+
+	// ticksOffset is the number of ticks that match "r" pixels
+	int ticksOffset = MidiTime::ticksPerBar() * r / m_ppb;
+
+	while (it != tm.end())
+	{
+		// If the x coordinate is within "r" pixels of the node's position
+		if (posTicks >= it.key() - ticksOffset)
+		{
+			if (posTicks <= it.key() + ticksOffset)
+			{
+				// The y position of the node
+				float valueY = yCoordOfLevel(
+					outValue
+						? it.value().getOutValue()
+						: it.value().getInValue()
+					);
```

Comment:
```suggestion
				float valueY = yCoordOfLevel(
					outValue
					? it.value().getOutValue()
					: it.value().getInValue()
				);
```

Reply:
Aren't the tabs for multiple line ternary operators required by the coding convention? It isn't 100% clear on it to be fair, it says it "Should be formatted as below" and have this as an example:

```
a == condition
	? value
	: otherValue;
```

Now I'm unsure whether the requirement is for them to be in separate lines or if the indentation is part of it as well.

Reply:
Ah, I see. The tabs shouldn't be required unless needed for clarity. We'll have to fix the wording slightly.